src/futures.rs

```diff
@@ -27,3 +27,23 @@ impl<T, F: Fn() -> Option<T>> Future for WaitForValue<F> {
         }
     }
 }
+
+/// Generates a synchronous `main()` function that calls the provided
+/// asynchronous function. To avoid name conflicts, the asynchronous function
+/// must not be called `main()`; `async_main()` is fine.
+///
+/// Example:
+///     libtock::async_main!(async_main);
+///     async fn async_main() {
+///         // Snipped
+///     }
+#[macro_export]
+macro_rules! async_main {
+    ($main_name:ident) => {
+        fn main() {
+            unsafe {
+                ::core::executor::block_on($main_name());
```

Comment:
So, the `Result` will be ignored? Won't this trigger a warning? We could call `unwrap`, s.t. code that doesn't exit with `Ok` leads to a panic.

Reply:
I tried to add an unwrap call, but that requires the error type to implement core::fmt::Debug. Unfortunately, adding Debug implementations has a large cost, so I don't want to use unwrap.

I agree we should handle this case but doing so is nontrivial, and I think it is outside the scope of this PR (which is strictly to allow us to update to libtock-rs master).

Reply:
You are right. `unwrap` is the wrong function. It should be `ok().unwrap()`.

Reply:
I added printing for the case where `Result::Err` is returned.